sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuSemaphore.scala

```diff
@@ -245,6 +259,30 @@ private final class SemaphoreTaskInfo() extends Logging {
     }
   }
 
+  def tryAcquire(semaphore: Semaphore): Boolean = synchronized {
```

Comment:
Do you think there are some smarter heuristics to be had if we return the number of waiters when the semaphore cannot be acquired via `Either[Int,Boolean]` ?